rules-reviewed/technology-usage/tests/data/http/JaxRsWebService.java

```diff
@@ -0,0 +1,51 @@
+package org.windup.examples.technology.http;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+
+@Path("/calc")
+public class JaxRsWebService extends AbstractEaoService implements Serializable {
```

Comment:
There are 5 identified `RxJSWebServices` in this file. For each `@Path` annotation. But is it correct? Shouldn't there be 4? The top-level one doesn't add any new service, it is just root for other services.

Reply:
That is the question why the https://github.com/windup/windup/blob/master/rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJaxRsAnnotationsRuleProvider.java#L43 is then constructed like it is. The condition correspond to the result. If you would like to have condition to check  effectively only real paths you should limit the annotation on methods and not for whole class I think.